connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -255,8 +255,8 @@ public ConfigInfos validateConnectorConfig(Map<String, String> connectorConfig)
             allGroups.addAll(basicConfigDef.groups());
 
             ConnectorConfig connectorConfigToEnrich = (connector instanceof SourceConnector)
-                    ? new SourceConnectorConfig(plugins(), connectorConfig)
-                    : new SinkConnectorConfig(plugins(), connectorConfig);
+                    ? new SourceConnectorConfig(plugins(), connectorConfig, false)
```

Comment:
I considered this approach as well, but I think we can just drastically simplify this. I'm not sure there was originally a great reason for checking the base config, then connector config. I think this was an artifact of the two parts being split up originally (before we even had enrichment for transformations). We still need 2 parts since the connector may do additional validation, but we should only need to get everything we need together for the framework-level validation into one ConfigDef and then `validateBasicConnectorConfig` on that.

I think a better approach might be to just make it easy to get the `SourceConnectorConfig.configDef()` and `SinkConnectorConfig.configDef()`, i.e. just make those public. Then you never need to instantiate the ConnectorConfig directly here. Instead, you use `ConnectorConfig.enrich()` on the `Source/SinkConnectorConfig.configDef()` as the baseConfigDef with the `requireFullConfig` parameter as `false`. In this way, this config never needs to instantiate the `ConnectorConfig` classes and you can leave their constructors as is.

I think this has some other nice benefits, including that we don't *have* to skip the connector-specific validation even if there are errors during the basic validation. I'm not sure we want to make that change, but it would mean that even if you have a validation error (e.g. you are missing a connector name), you can still get connector-specific fields back in the result, which seems like a good thing to me since you'd still want to know about those fields as long as you had entered the `connector.class` correctly.